public/components/datasources/components/manage/associated_objects/utils/associated_objects_tab_loading.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiEmptyPrompt, EuiLoadingSpinner } from '@elastic/eui';
+import React from 'react';
+
+interface AssociatedObjectsTabLoadingProps {
+  objectType: string;
+  warningMessage: boolean;
+}
+
+export const AssociatedObjectsTabLoading: React.FC<AssociatedObjectsTabLoadingProps> = (props) => {
+  const { objectType, warningMessage } = props;
+
+  const BodyText = () => (
```

Comment:
if all you need is a piece of static dom then you don't have to make it as a function